api/api/src/main/scala/uk/ac/wellcome/platform/api/services/ElasticsearchRequestBuilder.scala

```diff
@@ -1,159 +1,12 @@
-    .map {
-      case ProductionDateSortRequest => "data.production.dates.range.from"
-    }
-    .map { FieldSort(_).order(sortOrder) }
-
-  lazy val sortOrder = queryOptions.sortOrder match {
-    case SortingOrder.Ascending  => SortOrder.ASC
-    case SortingOrder.Descending => SortOrder.DESC
-  }
-
-  lazy val postFilterQuery: BoolQuery = boolQuery.filter {
-    filteredAggregationBuilder.pairedFilters.map(buildWorkFilterQuery)
-  }
-
-  lazy val filteredQuery: BoolQuery = queryOptions.searchQuery
-    .map { searchQuery =>
-      ElasticsearchComboQuery(searchQuery).elasticQuery
-    }
-    .getOrElse { boolQuery }
-    .filter {
-      (IdentifiedWorkFilter :: filteredAggregationBuilder.unpairedFilters)
-        .map(buildWorkFilterQuery)
-    }
-
-  private def buildWorkFilterQuery(workFilter: WorkFilter): Query =
-    workFilter match {
-      case IdentifiedWorkFilter =>
-        termQuery(field = "type", value = "IdentifiedWork")
-      case ItemLocationTypeFilter(itemLocationTypeIds) =>
-        termsQuery(
-          field = "data.items.locations.locationType.id",
-          values = itemLocationTypeIds)
-      case WorkTypeFilter(workTypeIds) =>
-        termsQuery(field = "data.workType.id", values = workTypeIds)
-      case DateRangeFilter(fromDate, toDate) =>
-        val (gte, lte) =
-          (fromDate map ElasticDate.apply, toDate map ElasticDate.apply)
-        RangeQuery("data.production.dates.range.from", lte = lte, gte = gte)
-      case LanguageFilter(languageIds) =>
-        termsQuery(field = "data.language.id", values = languageIds)
-      case GenreFilter(genreQuery) =>
-        simpleStringQuery(genreQuery)
-          .field("data.genres.label")
-          .defaultOperator("AND")
-      case SubjectFilter(subjectQuery) =>
-        simpleStringQuery(subjectQuery)
-          .field("data.subjects.label")
-          .defaultOperator("AND")
-      case LicenseFilter(licenseIds) =>
-        termsQuery(
-          field = "data.items.locations.license.id",
-          values = licenseIds)
-      case CollectionPathFilter(path) =>
-        termQuery(field = "data.collectionPath.path", value = path)
-      case CollectionDepthFilter(depth) =>
-        termQuery(field = "data.collectionPath.depth", value = depth)
-
-    }
-
-  implicit class EnhancedTermsAggregation(agg: TermsAggregation) {
-    def additionalField(field: String): TermsAggregation =
-      additionalFields(List(field))
-    def additionalFields(fields: List[String]): TermsAggregation = {
-      agg.subAggregations(
-        TopHitsAggregation("sample_doc")
-          .size(1)
-          .fetchSource(fields.toArray ++ agg.field, Array())
-      )
-    }
-  }
+import com.sksamuel.elastic4s.requests.searches.sort.FieldSort
+import com.sksamuel.elastic4s.Index
+
+trait ElasticsearchRequestBuilder {
+  def request(queryOptions: ElasticsearchQueryOptions,
+              index: Index,
+              scored: Boolean): SearchRequest
+  val idSort: FieldSort
```

Comment:
There is also a `CollectionRequestBuilder`, not sure whether fits this interface or not but potentially could extend from this too

Reply:
Good point, I'll see whether that can be done

Reply:
Unfortunately this doesn't really work and I'm not sure it's worth trying to make it fit